packages/extension/background.js

```diff
@@ -50,14 +50,50 @@ chrome.contextMenus.onClicked.addListener(async (info, tab) => {
 });
 
 chrome.action.onClicked.addListener((tab) => {
-  chrome.scripting.executeScript({
-    target: { tabId: tab.id, allFrames: false },
-    func: createPanel,
-  });
+  getEmailCheck(tab)
+  .then((email) => {
+    chrome.scripting.executeScript({
+      target: { tabId: tab.id, allFrames: false },
+      args: [email],
+      func: createPanel,
+    });
+  })
 });
 
 chrome.runtime.onMessage.addListener((message) => {
   if (message.type === "save-prefs") {
   } else if (message.type === "get-defs") {
   }
 });
+
+
+const getEmailCheck = (tab) => {
+  return new Promise((resolve, reject) => {
+    // Get email if we know where to look
+    if (tab.url.includes("zendesk")) {
+      chrome.scripting.executeScript({
+        target: { tabId: tab.id, allFrames: false },
+        func: () => {
+          let email = "";
+          const userEmailElement = document.getElementsByClassName("email-text")
+          if (userEmailElement.length > 0) {
+            email = userEmailElement[0].innerText
+          }
+
+          if (!email) {
+            const ticketEmail = document.getElementsByClassName("sc-7acs88-3 hljQCC")
```

Comment:
This fine for now, but in my experience usually these random-looking classes are auto-generated and can change at any point. In the future, we can either:
1. Use [XPath](https://developer.mozilla.org/en-US/docs/Web/XPath) to select based on DOM elements
2. Try to select the same element based on named classes

Nothing actionable at this point, but wanted to share some options for going forward.